xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererMediaCodecSurface.cpp

```diff
@@ -156,6 +156,17 @@ bool CRendererMediaCodecSurface::RenderUpdateVideoHook(bool clear, DWORD flags,
     mci->RenderUpdate(srcRect, dstRect);
   }
 
+  /** @todo This is a really bad hack and not meant to be
+  * a solution. This needs proper fixing when android maintainer
+  * is back. Real solution: Block until your next vblank.
```

Comment:
2) is "wait((1/fps) - (current_time - m_lastRenderTime))", right?
3) Where / how do you let player know? This is actually trivial, because you can assume it's the time of ::RenderUpdate() with an error of 1 frame at worst